src/components/TheTimer.vue

```diff
@@ -0,0 +1,344 @@
+        }
+    }
+
+    function playTimer(): void {
+        startTimer();
+        loadSong();
+        playSong();
+    }
+
+    function playSong(): void {
+        startSound.play();
+        if (!isStart.value) {
+            timerSound.pause();
+            timerSound.loop = false;
+        } else if (buttons.value[0].active) {
+            if (!(overTime.value === 0 && overSecond.value === "00")) {
+                timerSound.play();
+                timerSound.loop = true;
+            }
+        }
+    }
+
+    function deleteTask(id: number): void {
+        tasks.value = tasks.value.filter((item: Task) => item.id !== id);
+    }
+
+    function clearFinishedTasks(): void {
+        tasks.value = tasks.value.filter((item: Task) => !item.active);
+    }
+
+    function clearAllTasks(): void {
+        tasks.value = [];
+        finishedPomos.value = 1;
+    }
+
+    function onSubmit(obj: any): void {
+        if (obj.type === "add") {
+            tasks.value.push(
+                { id: Date.now(), work: obj.taskName, title: obj.taskNote, count: obj.taskRepeatCount, finishedCount: 0, active: false, isEdit: true, completed: false },
+            );
+        } else {
+            tasks.value.forEach((task: Task) => {
+                if (task.id === obj?.id) {
+                    task.work = obj.taskName;
+                    task.title = obj.taskNote;
+                    task.count = obj.taskRepeatCount;
+                }
+                task.isEdit = true;
+            });
+        }
+    }
+
+    function closeSettings(): void {
+        settingsShown.value = false;
+    }
+
+    function timeSaveChanges(obj: SettingTimer): void {
+        if (obj.breakLongTime && obj.breakShortTime && obj.pomodoroTime) {
+            buttons.value[0].time = obj.pomodoroTime;
+            buttons.value[1].time = obj.breakShortTime;
+            buttons.value[2].time = obj.breakLongTime;
+        }
+    }
+
+    function autoStartPomodoro(): void {
+        isStartPomodoros.value = !isStartPomodoros.value;
+    }
+
+    function autoStartBreaks(): void {
+        isStartBreaks.value = !isStartBreaks.value;
+    }
+
+    watch(overSecond, () => {
+        finishedTasks();
+        if (settingsShown.value) {
+            refreshTimer();
+            buttons.value.forEach((item: Button) => {
+                if (item.active) {
+                    bgColor.value = item.color;
+                }
+            });
+        }
+    });
+</script>
+
+<template>
+    <TheSettingsDialog
+        :show="settingsShown"
+        :is-start-pomodoros="isStartPomodoros"
+        :is-start-breaks="isStartBreaks"
+        @close="closeSettings"
+        @time-save-changes="timeSaveChanges"
+        @auto-start-pomodoro="autoStartPomodoro"
+        @auto-start-breaks="autoStartBreaks"
+    />
+    <div class="w-full text-center px-2 mx-auto max-w-[620px]">
+        <div class="h-2px rounded-full bg-white mb-[40px]" :style="{ width: `${borderW}%` }" />
```

Comment:
<TheSettingsDialog>
    <template name="timer">
        <SettingsDialogTimer
            :is-start-pomodoros="isStartPomodoros"
            :is-start-breaks="isStartBreaks"
            @time-save-changes="timeSaveChanges"
            @auto-start-pomodoro="autoStartPomodoro"
            @auto-start-breaks="autoStartBreaks"
        />
    </template>
</TheSettingsDialog>

shunaqa qisayiz settings dialogga keyin emitlani oshirmasdan togridan togri SettingsDialogTimerga bersa bo'ladi, slotlani o'rganib chiqing vueni documentatsiyasidan, slot bn v-modelni qo'llashda qiynalyapsiz, slotladan actionlani chiqarvosayam boladi shulani ko'rib chiqing